test/unit/math/mix/mat/prob/std_binormal_lcdf_test.cpp

```diff
@@ -0,0 +1,1245 @@
+  int N_y_;
+  vector<double> rho_;
+  vV_vD_std_binorm_lcdf(int N_y, vector<double>& rho) : N_y_(N_y), rho_(rho) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, Dynamic, 1>> y;
+    to_function_input(N_y_, inp_vec, y);
+    return stan::math::std_binormal_lcdf(y, rho_);
+  }
+};
+
+// vector<Row Vector>, double vector
+struct vRV_vD_std_binorm_lcdf {
+  int N_y_;
+  vector<double> rho_;
+  vRV_vD_std_binorm_lcdf(int N_y, vector<double>& rho) : N_y_(N_y), rho_(rho) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, 1, Dynamic>> y;
+    to_function_input(N_y_, inp_vec, y);
+    return stan::math::std_binormal_lcdf(y, rho_);
+  }
+};
+
+// vector<Vector>, Vector
+struct vV_VD_std_binorm_lcdf {
+  int N_y_;
+  Matrix<double, Dynamic, 1> rho_;
+  vV_VD_std_binorm_lcdf(int N_y, Matrix<double, Dynamic, 1>& rho)
+      : N_y_(N_y), rho_(rho) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, Dynamic, 1>> y;
+    to_function_input(N_y_, inp_vec, y);
+    return stan::math::std_binormal_lcdf(y, rho_);
+  }
+};
+
+// vector<Row Vector>, Vector
+struct vRV_VD_std_binorm_lcdf {
+  int N_y_;
+  Matrix<double, Dynamic, 1> rho_;
+  vRV_VD_std_binorm_lcdf(int N_y, Matrix<double, Dynamic, 1>& rho)
+      : N_y_(N_y), rho_(rho) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, 1, Dynamic>> y;
+    to_function_input(N_y_, inp_vec, y);
+    return stan::math::std_binormal_lcdf(y, rho_);
+  }
+};
+
+// double Vector, real
+struct VD_R_std_binorm_lcdf {
+  Matrix<double, Dynamic, 1> y_;
+  explicit VD_R_std_binorm_lcdf(Matrix<double, Dynamic, 1>& y) : y_(y) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    return stan::math::std_binormal_lcdf(y_, inp_vec(0));
+  }
+};
+
+// double Row Vector, real
+struct RVD_R_std_binorm_lcdf {
+  Matrix<double, 1, Dynamic> y_;
+  explicit RVD_R_std_binorm_lcdf(Matrix<double, 1, Dynamic>& y) : y_(y) {}
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    return stan::math::std_binormal_lcdf(y_, inp_vec(0));
+  }
+};
+
+// double vector<Vector>, real
+struct vVD_R_std_binorm_lcdf {
+  vector<Matrix<double, Dynamic, 1>> y_;
+  explicit vVD_R_std_binorm_lcdf(vector<Matrix<double, Dynamic, 1>>& y) {
+    for (size_t i = 0; i < y.size(); ++i)
+      y_.push_back(y[i]);
+  }
+  template <typename T>
+  inline T operator()(
+      const Eigen::Matrix<T, Eigen::Dynamic, 1>& inp_vec) const {
+    vector<Matrix<T, Dynamic, 1>> y;
+    return stan::math::std_binormal_lcdf(y_, inp_vec(0));
+  }
+};
+
+// double vector<Row Vector>, real
```

Comment:
Basically, way too many of these.  Figure out how to consolidate using templates if at all possible.

If not, document the scheme up top.